src/shogun/multiclass/tree/FeatureImportanceTree.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+
+#ifndef _FEATUREIMPORTANCETREE_H__
+#define _FEATUREIMPORTANCETREE_H__
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/multiclass/tree/TreeMachine.h>
+#include <shogun/multiclass/tree/TreeMachineNode.h>
+
+namespace shogun
+{
+
+	/** @brief class FeatureImportanceTree, a base class for the tree which need
+	 * computing feature importances. This class is derived from TreeMachine<T>
+	 * and stores the feature importances
+	 *
+	 */
+	template <typename T>
+	class FeatureImportanceTree : public TreeMachine<T>
```

Comment:
Maybe `FeatureImportanceMixin` would be a better name?
@gf712 @karlnapf ?

Reply:
Yes, and change the typename to something meaningful, e.g TreeType

Reply:
I think having “mixin” in the name is not necessary as it is bulky, would rather put that in the api docstring. But +1 for a meaningful type name